src/WebJobs.Extensions/Extensions/Warmup/Trigger/WarmupTriggerBinding.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Warmup.Trigger
+{
+    internal class WarmupTriggerBinding : ITriggerBinding
+    {
+        private readonly ParameterInfo _parameter;
+
+        public WarmupTriggerBinding(ParameterInfo parameter)
+        {
+            _parameter = parameter;
+        }
+
+        public Type TriggerValueType
+        {
+            get { return _parameter.ParameterType; }
+        }
+
+        public IReadOnlyDictionary<string, Type> BindingDataContract => null;
+
+        public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+        {
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            Utility.ApplyBindingData(value, bindingData);
```

Comment:
Your BindingDataContract is null above (which makes sense for this trigger I think), so why are you applying binding data here? I think for this trigger we don't need any binding data/contract. 

Manual trigger is different, because users own the POCO being passed in, and may use it to define values that they want to bind to (part of binding data) in other input/output bindings.

Reply:
You are right. I think I mixed this up. My thought was initially to not have a contract. Then I thought maybe users can bind to certain elements and not the entire trigger input (if we were to expand the context of this trigger). But that does not make sense in this context. And if we were to do something like that, I should have had a contract.

Good catch! I will update.